### PR TITLE
Fix Clang-tidy readability-redundant-member-init warnings. Modernize > >

### PR DESCRIPTION
**Pull request by EugeneZelenko**

### CODE REVIEW

File: iwyu_output.cc [hunk 1]
```diff
@@ -240,11 +240,7 @@ OneUse::OneUse(const NamedDecl* decl, SourceLocation use_loc,
       use_loc_(use_loc),
       use_kind_(use_kind),             // full use or fwd-declare use
       in_cxx_method_body_(in_cxx_method_body),
-      comment_(comment ? comment : ""),
-      public_headers_(),
-      suggested_header_(),             // figure that out later
-      ignore_use_(false),
-      is_iwyu_violation_(false) {
```
Comment (kimgr) [inline iwyu_output.cc]: We shouldn't be removing `ignore_use_` and `is_iwyu_violation_` here, should we?

Won't they be garbage? I think I've seen recent claims that members are default-initialized, but I can't find a reference now. I'd prefer to keep the explicitly initialized to false anyway.

File: iwyu_output.cc [hunk 2]
```diff
@@ -257,12 +253,7 @@ OneUse::OneUse(const string& symbol_name, const FileEntry* dfn_file,
       decl_filepath_(dfn_filepath),
       use_loc_(use_loc),
       use_kind_(kFullUse),
-      in_cxx_method_body_(false),
-      comment_(),
-      public_headers_(),
-      suggested_header_(),
-      ignore_use_(false),
-      is_iwyu_violation_(false) {
```
Comment (kimgr) [inline iwyu_output.cc]: Keep Boolean inits

File: iwyu_output.cc [hunk 3]
```diff
@@ -434,9 +425,10 @@ string MungedForwardDeclareLine(const NamedDecl* decl) {
 OneIncludeOrForwardDeclareLine::OneIncludeOrForwardDeclareLine(
     const NamedDecl* fwd_decl)
     : line_(internal::MungedForwardDeclareLine(fwd_decl)),
-      start_linenum_(-1), end_linenum_(-1),     // set 'for real' below
-      is_desired_(false), is_present_(false), symbol_counts_(),
```
Comment (kimgr) [inline iwyu_output.cc]: Keep Boolean inits

File: iwyu_output.cc [hunk 4]
```diff
@@ -448,9 +440,10 @@ OneIncludeOrForwardDeclareLine::OneIncludeOrForwardDeclareLine(
 OneIncludeOrForwardDeclareLine::OneIncludeOrForwardDeclareLine(
     const FileEntry* included_file, const string& quoted_include, int linenum)
     : line_("#include " + quoted_include),
-      start_linenum_(linenum), end_linenum_(linenum),
-      is_desired_(false), is_present_(false), symbol_counts_(),
```
Comment (kimgr) [inline iwyu_output.cc]: Keep Boolean inits

File: iwyu_output.h [hunk 5]
```diff
@@ -108,8 +108,8 @@ class OneUse {
   string comment_;                 // If not empty, append to clang warning msg
   vector<string> public_headers_;  // header to #include if dfn hdr is private
   string suggested_header_;        // header that allows us to satisfy use
-  bool ignore_use_;                // set to true if use is discarded
-  bool is_iwyu_violation_;         // set to false when we figure out it's not
+  bool ignore_use_ = false;        // set to true if use is discarded
```
Comment (kimgr) [inline iwyu_output.h]: Ah, I see what happened now. Can we please keep initialization in the init-list? Better to have initialization in one place.

File: iwyu_output.h [hunk 6]
```diff
@@ -108,8 +108,8 @@ class OneUse {
...
+  bool is_iwyu_violation_ = false; // set to false when we figure out it's not
```
Comment (kimgr) [inline iwyu_output.h]: Move back to init list

File: iwyu_output.h [hunk 7]
```diff
@@ -169,8 +169,8 @@ class OneIncludeOrForwardDeclareLine {
   string line_;                     // '#include XXX' or 'class YYY;'
   int start_linenum_;
   int end_linenum_;
-  bool is_desired_;                 // IWYU will recommend this line
-  bool is_present_;                 // line was present before the IWYU run
+  bool is_desired_ = false;         // IWYU will recommend this line
```
Comment (kimgr) [inline iwyu_output.h]: Move back to init list

**Comment by EugeneZelenko** [comment]
I reverted C++11-style member initialization. I also fixed Clang-tidy modernize-use-default warnings.

**Comment by kimgr** [comment]
Looks good to me. @vsapsai ?

**Comment by vsapsai** [comment]
Looks good to me and there are no regressions. My knowledge is outdated but as long as MSVC version we are using supports defaulted constructors and destructors, I have no objections.

**Comment by EugeneZelenko** [comment]
Minimum MSVC version required by LLVM should support = default. Problem was some default constructors/destructors with GCC 4.7, but requirement was changed to GCC 4.8.

**Comment by kimgr** [comment]
Let me double-check MSVC (Windows) and clang-3.6 builds (FreeBSD). I'll merge if it goes well.

**Comment by kimgr** [comment]
Done, eb7f5d1ef02836. Thanks!

**Pull request closed without merging**